diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2dff438..cf8d3ad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,7 +22,7 @@ endif()
 
 list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
@@ -139,11 +139,11 @@ endif()
 target_include_directories(franka PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include>
+  ${Poco_INCLUDE_DIRS}
 )
 
 target_link_libraries(franka PRIVATE
-  Poco::Foundation
-  Poco::Net
+  ${Poco_LIBRARIES}
   Eigen3::Eigen3
   Threads::Threads
   pinocchio::pinocchio
@@ -202,13 +202,13 @@ install(EXPORT FrankaTargets
 # Ignore find_package(Franka) in subprojects.
 set(FRANKA_IS_FOUND TRUE)
 
-option(BUILD_TESTS "Build tests" ON)
+option(BUILD_TESTS "Build tests" OFF)
 if(BUILD_TESTS)
   enable_testing()
   add_subdirectory(test)
 endif()
 
-option(BUILD_EXAMPLES "Build example code" ON)
+option(BUILD_EXAMPLES "Build example code" OFF)
 if(BUILD_EXAMPLES)
   add_subdirectory(examples)
 endif()
diff --git a/cmake/FindPoco.cmake b/cmake/FindPoco.cmake
index 36ef224..fbbdc35 100644
--- a/cmake/FindPoco.cmake
+++ b/cmake/FindPoco.cmake
@@ -1,32 +1,185 @@
-find_package(Poco COMPONENTS ${Poco_FIND_COMPONENTS} CONFIG QUIET)
-if(Poco_FOUND)
-  return()
-endif()
-
-find_path(Poco_INCLUDE_DIR Poco/Poco.h)
-mark_as_advanced(FORCE Poco_INCLUDE_DIR)
-
-foreach(component ${Poco_FIND_COMPONENTS})
-  set(component_var "Poco_${component}_LIBRARY")
-  find_library(${component_var} Poco${component})
-  mark_as_advanced(FORCE ${component_var})
-  if(${component_var})
-    set(Poco_${component}_FOUND TRUE)
-    list(APPEND Poco_LIBRARIES ${component})
-    if(NOT TARGET Poco::${component})
-      add_library(Poco::${component} SHARED IMPORTED)
-      set_target_properties(Poco::${component} PROPERTIES
-        INTERFACE_INCLUDE_DIRECTORIES ${Poco_INCLUDE_DIR}
-        IMPORTED_LOCATION ${${component_var}}
-      )
-    endif()
-  endif()
-endforeach()
-
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(Poco
-  FOUND_VAR Poco_FOUND
-  REQUIRED_VARS Poco_INCLUDE_DIR Poco_LIBRARIES
-  VERSION_VAR Poco_VERSION
-  HANDLE_COMPONENTS
+# - Find the Poco includes and libraries.
+# The following variables are set if Poco is found.  If Poco is not
+# found, Poco_FOUND is set to false.
+#  Poco_FOUND        - True when the Poco include directory is found.
+#  Poco_INCLUDE_DIRS - the path to where the poco include files are.
+#  Poco_LIBRARY_DIR - The path to where the poco library files are.
+#  Poco_BINARY_DIRS - The path to where the poco dlls are.
+#  Poco_LIBRARIES - list of all libs from requested components.
+
+# ----------------------------------------------------------------------------
+# If you have installed Poco in a non-standard location.
+# Then you have three options.
+# In the following comments, it is assumed that <Your Path>
+# points to the root directory of the include directory of Poco. e.g
+# If you have put poco in C:\development\Poco then <Your Path> is
+# "C:/development/Poco" and in this directory there will be two
+# directories called "include" and "lib".
+# 1) After CMake runs, set Poco_INCLUDE_DIR to <Your Path>/poco<-version>
+# 2) Use CMAKE_INCLUDE_PATH to set a path to <Your Path>/poco<-version>. This will allow FIND_PATH()
+#    to locate Poco_INCLUDE_DIR by utilizing the PATH_SUFFIXES option. e.g.
+#    SET(CMAKE_INCLUDE_PATH ${CMAKE_INCLUDE_PATH} "<Your Path>/include")
+# 3) Set an environment variable called ${POCO_ROOT} that points to the root of where you have
+#    installed Poco, e.g. <Your Path>. It is assumed that there is at least a subdirectory called
+#    Foundation/include/Poco in this path.
+#
+# Note:
+#  1) If you are just using the poco headers, then you do not need to use
+#     Poco_LIBRARY_DIR in your CMakeLists.txt file.
+#  2) If Poco has not been installed, then when setting Poco_LIBRARY_DIR
+#     the script will look for /lib first and, if this fails, then for /stage/lib.
+#
+# Usage:
+# In your CMakeLists.txt file do something like this:
+# ...
+# # Poco
+# FIND_PACKAGE(Poco COMPONENTS XML Net Data...)
+# ...
+# INCLUDE_DIRECTORIES(${Poco_INCLUDE_DIRS})
+# LINK_DIRECTORIES(${Poco_LIBRARY_DIR})
+#
+# In Windows, we make the assumption that, if the Poco files are installed, the default directory
+# will be C:\poco or C:\Program Files\Poco or C:\Programme\Poco.
+
+MESSAGE(STATUS "Searching for Poco library...")
+
+SET(POCO_INCLUDE_PATH_DESCRIPTION "top-level directory containing the poco include directories. E.g /usr/local/include/ or c:\\poco\\include\\poco-1.3.2")
+SET(POCO_INCLUDE_DIR_MESSAGE "Set the Poco_INCLUDE_DIR cmake cache entry to the ${POCO_INCLUDE_PATH_DESCRIPTION}")
+SET(POCO_LIBRARY_PATH_DESCRIPTION "top-level directory containing the poco libraries.")
+SET(POCO_LIBRARY_DIR_MESSAGE "Set the Poco_LIBRARY_DIR cmake cache entry to the ${POCO_LIBRARY_PATH_DESCRIPTION}")
+
+
+SET(POCO_DIR_SEARCH $ENV{POCO_ROOT})
+IF(POCO_DIR_SEARCH)
+  FILE(TO_CMAKE_PATH ${POCO_DIR_SEARCH} POCO_DIR_SEARCH)
+ENDIF(POCO_DIR_SEARCH)
+
+
+# Add in some path suffixes. These will have to be updated whenever a new Poco version comes out.
+SET(SUFFIX_FOR_INCLUDE_PATH
+ poco-1.3.2
+ poco-1.3.3
+ poco-1.3.4
+ poco-1.3.5
+ poco-1.3.6
+)
+
+SET(SUFFIX_FOR_LIBRARY_PATH
+ poco-1.3.2/lib
+ poco-1.3.2/lib/Linux/i686
+ poco-1.3.2/lib/Linux/x86_64
+ poco-1.3.3/lib
+ poco-1.3.3/lib/Linux/i686
+ poco-1.3.3/lib/Linux/x86_64
+ poco-1.3.4/lib
+ poco-1.3.4/lib/Linux/i686
+ poco-1.3.4/lib/Linux/x86_64
+ poco-1.3.5/lib
+ poco-1.3.5/lib/Linux/i686
+ poco-1.3.5/lib/Linux/x86_64
+ poco-1.3.6/lib
+ poco-1.3.6/lib/Linux/i686
+ poco-1.3.6/lib/Linux/x86_64
+ lib
+ lib/Linux/i686
+ lib/Linux/x86_64
+)
+
+#
+# Look for an installation.
+#
+FIND_PATH(Poco_INCLUDE_DIR NAMES Foundation/include/Poco/SharedLibrary.h PATH_SUFFIXES ${SUFFIX_FOR_INCLUDE_PATH} PATHS
+
+# Look in other places.
+  ${POCO_DIR_SEARCH}
+
+# Help the user find it if we cannot.
+  DOC "The ${POCO_INCLUDE_PATH_DESCRIPTION}"
 )
+
+IF(NOT Poco_INCLUDE_DIR)
+
+# Look for standard unix include paths
+  FIND_PATH(Poco_INCLUDE_DIR Poco/Poco.h DOC "The ${POCO_INCLUDE_PATH_DESCRIPTION}")
+
+ENDIF(NOT Poco_INCLUDE_DIR)
+
+# Assume we didn't find it.
+SET(Poco_FOUND 0)
+
+# Now try to get the include and library path.
+IF(Poco_INCLUDE_DIR)
+  IF(EXISTS "${Poco_INCLUDE_DIR}/Foundation/include/Poco/SharedLibrary.h")
+    SET(Poco_INCLUDE_DIRS
+      ${Poco_INCLUDE_DIR}/CppUnit/include
+      ${Poco_INCLUDE_DIR}/Foundation/include
+      ${Poco_INCLUDE_DIR}/Net/include
+      ${Poco_INCLUDE_DIR}/Util/include
+      ${Poco_INCLUDE_DIR}/XML/include
+    )
+    SET(Poco_FOUND 1)
+  ELSEIF(EXISTS "${Poco_INCLUDE_DIR}/Poco/Poco.h")
+    SET(Poco_INCLUDE_DIRS
+      ${Poco_INCLUDE_DIR}
+    )
+    SET(Poco_FOUND 1)
+  ENDIF()
+
+  IF(NOT Poco_LIBRARY_DIR)
+
+    FIND_LIBRARY(Poco_FOUNDATION_LIB NAMES PocoFoundation PocoFoundationd  PATH_SUFFIXES ${SUFFIX_FOR_LIBRARY_PATH} PATHS
+
+# Look in other places.
+      ${Poco_INCLUDE_DIR}
+      ${POCO_DIR_SEARCH}
+
+# Help the user find it if we cannot.
+      DOC "The ${POCO_LIBRARY_PATH_DESCRIPTION}"
+    )
+    SET(Poco_LIBRARY_DIR "" CACHE PATH POCO_LIBARARY_PATH_DESCRIPTION)
+    GET_FILENAME_COMPONENT(Poco_LIBRARY_DIR ${Poco_FOUNDATION_LIB} PATH)
+    SET(Poco_LIBRARIES "")
+    SET(Comp_List "")
+    IF(Poco_LIBRARY_DIR AND Poco_FOUNDATION_LIB)
+# Look for the poco binary path.
+      SET(Poco_BINARY_DIR ${Poco_INCLUDE_DIR})
+      IF(Poco_BINARY_DIR AND EXISTS "${Poco_BINARY_DIR}/bin")
+        SET(Poco_BINARY_DIRS ${Poco_BINARY_DIR}/bin)
+      ENDIF(Poco_BINARY_DIR AND EXISTS "${Poco_BINARY_DIR}/bin")
+    ENDIF(Poco_LIBRARY_DIR AND Poco_FOUNDATION_LIB)
+
+    IF(Poco_FOUNDATION_LIB)
+      FOREACH(COMPONENT ${Poco_FIND_COMPONENTS})
+          FIND_LIBRARY(LIB${COMPONENT} "Poco${COMPONENT}" Poco_LIBRARY_DIR)
+          IF (LIB${COMPONENT})
+              LIST(APPEND Poco_LIBRARIES "${LIB${COMPONENT}}")
+              LIST(APPEND Comp_List "${COMPONENT}")
+          ELSE ()
+              IF (${Poco_FIND_REQUIRED})
+                MESSAGE(SEND_ERROR "Poco component ${COMPONENT} NOT FOUND!")
+              ENDIF ()
+          ENDIF(LIB${COMPONENT})
+      ENDFOREACH(COMPONENT)
+      LIST(REMOVE_DUPLICATES Comp_List)
+    ENDIF(Poco_FOUNDATION_LIB)
+  ENDIF(NOT Poco_LIBRARY_DIR)
+ENDIF(Poco_INCLUDE_DIR)
+
+IF(NOT Poco_FOUND)
+  IF(Poco_FIND_QUIETLY)
+    MESSAGE(STATUS "Poco was not found. ${POCO_INCLUDE_DIR_MESSAGE}")
+  ELSE(Poco_FIND_QUIETLY)
+    IF(Poco_FIND_REQUIRED)
+      MESSAGE(FATAL_ERROR "Poco was not found. ${POCO_INCLUDE_DIR_MESSAGE}")
+    ENDIF(Poco_FIND_REQUIRED)
+  ENDIF(Poco_FIND_QUIETLY)
+ELSE(NOT Poco_FOUND)
+  MESSAGE(STATUS "  Found Poco!")
+  SET(COMPONENT_STR "components found:")
+  FOREACH(comp ${Comp_List})
+   SET(COMPONENT_STR "${COMPONENT_STR}, ${comp}")
+
+  ENDFOREACH(comp ${Comp_List})
+  STRING(REPLACE ":," ":" COMPONENT_LSTR ${COMPONENT_STR})
+  MESSAGE(STATUS "${COMPONENT_LSTR}.")
+ENDIF(NOT Poco_FOUND)
\ No newline at end of file
diff --git a/include/franka/logging/logger.hpp b/include/franka/logging/logger.hpp
index 93e0e80..2231465 100644
--- a/include/franka/logging/logger.hpp
+++ b/include/franka/logging/logger.hpp
@@ -37,52 +37,16 @@ auto removeAllLoggers() -> void;
  */
 auto logInfo(const std::string& message) -> void;
 
-/**
- * Logs an info message.
- * @tparam S The type of the format string.
- * @tparam Args The types of the arguments.
- * @param format_str The format string.
- * @param args The arguments.
- */
-template <typename S, typename... Args>
-auto logInfo(const S& format_str, Args&&... args) -> void {
-  logInfo(fmt::format(format_str, std::forward<Args>(args)...));
-}
-
 /**
  * Logs a warning message.
  * @param message The message to log.
  */
 auto logWarn(const std::string& message) -> void;
 
-/**
- * Logs a warning message.
- * @tparam S The type of the format string.
- * @tparam Args The types of the arguments.
- * @param format_str The format string.
- * @param args The arguments.
- */
-template <typename S, typename... Args>
-auto logWarn(const S& format_str, Args&&... args) -> void {
-  logWarn(fmt::format(format_str, std::forward<Args>(args)...));
-}
-
 /**
  * Logs an error message.
  * @param message The message to log.
  */
 auto logError(const std::string& message) -> void;
 
-/**
- * Logs an error message.
- * @tparam S The type of the format string.
- * @tparam Args The types of the arguments.
- * @param format_str The format string.
- * @param args The arguments.
- */
-template <typename S, typename... Args>
-auto logError(const S& format_str, Args&&... args) -> void {
-  logError(fmt::format(format_str, std::forward<Args>(args)...));
-}
-
 }  // namespace franka::logging
diff --git a/src/active_control.cpp b/src/active_control.cpp
index 7f0ae0d..143d31a 100644
--- a/src/active_control.cpp
+++ b/src/active_control.cpp
@@ -22,7 +22,7 @@ ActiveControl::ActiveControl(std::shared_ptr<Robot::Impl> robot_impl,
       control_finished(false) {
   auto warning_message = std::string();
   if (!setCurrentThreadToHighestSchedulerPriority(&warning_message)) {
-    logging::logWarn("{}", warning_message);
+    logging::logWarn(warning_message);
   }
 }
 
